paasta_tools/setup_kubernetes_cr.py

```diff
@@ -162,12 +163,19 @@ def setup_all_custom_resources(
         kube_client.apiextensions,
         kube_client.apiextensions_v1_beta1,
     ]:
-        cluster_crds = {
-            crd.spec.names.kind
-            for crd in apiextension.list_custom_resource_definition(
+
+        try:
+            crds_list = apiextension.list_custom_resource_definition(
                 label_selector=paasta_prefixed("service")
             ).items
-        }
+        except ApiException:
+            log.debug(
+                "Listing CRDs with apiextensions/v1beta1 not supported on this cluster, falling back to v1"
+            )
+            crds_list = []
+
+        cluster_crds = {crd.spec.names.kind for crd in crds_list}
+        succeeded = False
```

Comment:
should we be setting the initial value for `succeeded` outside this loop?

otherwise, a cluster that supports v1 and not v1beta1 will always return False, right?

Reply:
my mistake, yes you're right :)